### DISCUSSION TITLE
GWAT estimates

### DISCUSSION THREAD

**Discussion by Saman-M-B** [fabern/LWFBrook90.jl#89] [Q&A] [answered]
Hi, I was wondering if there is a function to retrieve the values of storage but in different depths such as GWAT or since the model considers these as storage, you can only get the final (deepest layer) values? Cause I think the LWFBrook90R model provides these values per soil layer for each timestep.

**Comment by fabern** [comment, marked as answer]
Hi Saman,

Sorry for the late reply. Your message has somehow escaped from my view.

Yes, since version 0.9.9 there is a way. This version introduced the functions `get_states`,  `get_fluxes` (and `get_forcing`). So for your request use `get_states` to get GWAT or SWAT from certain depths. This gives you tables with the storage or fluxes at each time step. Have a look at the documentation here: https://fabern.github.io/LWFBrook90.jl/stable/generated/example-script-01/#Postprocessing:-Extract-values

Hope this helps.

Best regards, Fabian

**Reply by Saman-M-B** [reply]
I checked but these 3 functions don't give me GWAT at each layer... they do for isotopes but not the rest

**Reply by fabern** [reply]
`get_states` should give you columns for `d18O`, `d2H`, but also volumetric water content `θ_m3m3_...` and soil moisture potential `ψ_kPa_...` for each simulated depth. From the water content you can then derive how many mm of soil water (SWAT) you have in the unsaturated zone by considering the thickness of the layer and the stone fraction.

GWAT itself is also output. Note that for GWAT there is only one compartment. So you cannot read out different depths. (I suppose what you want is SWAT.) Do these explanations help? What columns do you get when running `get_states`?

**Reply by Saman-M-B** [reply]
Thanks for your reply. I get the outputs you mentioned but I wanted to see GWAT per layer to see cumulatively how much water passes through in each time step for each layer. Not sure if it was the same but I remember when I first ran LWFBrook90R it had something like this.

**Reply by fabern** [reply]
Only the unsaturated zone (SWAT) is separated in layers. The saturated zone, however, is represented by a single compartment (GWAT) that lies underneath all the soil layers. (See fig 1 and 2 in the [documentation](https://fabern.github.io/LWFBrook90.jl/stable/model/)). This so the case in LWFBrook90R and the initial BROOK90 models.

**Reply by Saman-M-B** [reply]
Ok Thanks
